foundry/foundry_dataset.py

```diff
@@ -0,0 +1,248 @@
+import json
+import logging
+import os
+
+from json2table import convert
+import pandas as pd
+import numpy as np
+
+from foundry.models import FoundrySchema
+from pydantic import ValidationError
+
+
+logger = logging.getLogger(__name__)
+
+
+class FoundryDataset(FoundrySchema):
+    """Representation of an individual dataset.
+        Provides access to metadata as well as functions to
+        instantiate data into memory in different formats.
+
+        Desired functions:
+            - Get as pandas
+            - Get as tensorflow dataset
+            - Get as pytorch dataset
+            - Get file list
+            - Set metadata
+            - Attach datafiles
+            - Validate against schema
+            - Get citation
+        """
+
+    def __init__(self, source_id: str, datacite_entry: dict, foundry_schema: FoundrySchema):
+        self.source_id = source_id
+        self.dc = datacite_entry
+        self.foundry_schema = foundry_schema
+
+        return pd.DataFrame(
```

Comment:
I think it returns the metadata as a pandas dataframe so it can display nicely for the user, right? And then if we want it displayed as a list, that will be added in a param set up later? @blue442 let us know if we are understanding this correctly

Reply:
Yeah so this is just leftover cruff from some of the old `search` function that should've been / will be tossed - mainly kept around as a template for how we wanted to see the results from search displayed in a notebook (I think as @kjschmidt913 was referring to). 

Oddly enough, having a non-None return statement in the __init__ doesn't even appear to be valid python syntax, so I'm not really even sure why it's not exploding - I'm removing this.